### ISSUE TITLE
Update to support Java Modules Plugin version 1.6.0

### ISSUE THREAD

**Issue by msgilligan** [openjfx/javafx-gradle-plugin#59]
The [Java Module Plugin version 1.6.0](https://github.com/java9-modularity/gradle-modules-plugin/releases/tag/1.6.0) introduced some refactoring (see [gradle-modules-plugin Issue 80](https://github.com/java9-modularity/gradle-modules-plugin/issues/80)) that causes `java.lang.ClassNotFoundException: org.javamodularity.moduleplugin.tasks.ModuleOptions` -- see this [CI build log example](https://gitlab.com/ConsensusJ/bitcoinj/-/jobs/285098442).

**Comment by tlinkowski** [comment]
To update [org.javamodularity.moduleplugin](https://github.com/java9-modularity/gradle-modules-plugin) to [1.6.0](https://github.com/java9-modularity/gradle-modules-plugin/releases/tag/1.6.0), please replace `org.javamodularity.moduleplugin.tasks.ModuleOptions` with [`org.javamodularity.moduleplugin.extensions.RunModuleOptions`](https://github.com/java9-modularity/gradle-modules-plugin/blob/5e49c01f49a6e830c64b29f31eba0f064cf1d45d/src/main/java/org/javamodularity/moduleplugin/extensions/RunModuleOptions.java) here:
https://github.com/openjfx/javafx-gradle-plugin/blob/9727489a994bdccbac5b9d38c626f7efa61a27c2/src/main/java/org/openjfx/gradle/tasks/ExecTask.java#L38
and `ModuleOptions` with `RunModuleOptions` here:
https://github.com/openjfx/javafx-gradle-plugin/blob/9727489a994bdccbac5b9d38c626f7efa61a27c2/src/main/java/org/openjfx/gradle/tasks/ExecTask.java#L76

**Comment by msgilligan** [comment]
@johanvos Any chance this update can be made as part of (or shortly following) any updates for OpenJFX 13?

**Comment by jperedadnr** [comment]
@msgilligan, @tlinkowski, @yuliu2016, now that the PR #62 has been merged, a new snapshot of the plugin is available [here](https://oss.sonatype.org/content/repositories/snapshots/org/openjfx/javafx-gradle-plugin/0.0.9-SNAPSHOT/).

It would be great if you could test it and provide some feedback before we release it?

You know how to do it, but just in case, add this to your `settings.gradle` file:

```
pluginManagement {
    repositories {
        maven {
            url "https://oss.sonatype.org/content/repositories/snapshots"
        }
        gradlePluginPortal()
    }
}
```

**Comment by yuliu2016** [comment]
I've tried this out with my projects. The JavaFX plugin itself was fine and working, but the Modules plugin has some refactoring/breaking changes in version 1.6.0. Specifically, I had an issue with https://github.com/java9-modularity/gradle-modules-plugin/issues/118 (project doesn't work with the Kotlin plugin applied), but I worked around it by having another submodule (without the Kotlin plugin) depend on the Kotlin module.

**Comment by jperedadnr** [comment]
@yuliu2016 Thanks for testing. So nothing that can be fixed from the JavaFX gradle plugin, right?

**Comment by yuliu2016** [comment]
I don't think there's anything to be done

**Comment by msgilligan** [comment]
@jperedadnr I finally created a branch of my https://github.com/ConsensusJ/wallet-framework/ project that uses `0.0.9-SNAPSHOT` and Modules Plugin `1.6.0`. Note that I'm also using the Badass Jlink Plugin. Everything builds and runs correctly!

Here's a WIP PR with the changes: https://github.com/ConsensusJ/wallet-framework/pull/7 -- you can see that both Github Actions and GitLab CI were able to build the branch.

**Comment by msgilligan** [comment]
@tlinkowski Have you tested the snapshot? Any reason not to release it?

@jperedadnr Are you ready to release 0.0.9?

(I'm ready to upgrade my projects to 0.0.9 release and Modules plugin 1.6.0 😁)

**Comment by tlinkowski** [comment]
> @tlinkowski Have you tested the snapshot? Any reason not to release it?

No, I haven't.

Sorry, but I have never even used javafx-gradle-plugin. I came here from https://github.com/java9-modularity/gradle-modules-plugin/issues/80#issuecomment-526956120 to share an overview on how to update this plugin (as I know the internals of gradle-modules-plugin).

Unfortunately, gradle-modules-plugin has some major bugs (e.g. mentioned https://github.com/java9-modularity/gradle-modules-plugin/issues/118), but I won't find the time to fix them in the foreseeable future.

**Comment by msgilligan** [comment]
> Unfortunately, gradle-modules-plugin has some major bugs

Are those bugs present in 1.6.0 but not in 1.5.0?

**Comment by tlinkowski** [comment]
I'm not sure about the other bugs, but https://github.com/java9-modularity/gradle-modules-plugin/issues/118 (mentioned before) is almost certainly a regression introduced in 1.6.0 by https://github.com/java9-modularity/gradle-modules-plugin/issues/101.

**Comment by tobiasdiez** [comment]
Since the version 1.7 of the gradle modules plugin was just released, is there any chance 0.0.9 gets released soon as well?

**Comment by samypr100** [comment]
It seems 0.0.9 was released. I believe this can be closed.
